classes/class-wc-connect-taxjar-integration.php

```diff
@@ -1499,4 +1502,109 @@ public function load_taxjar_admin_new_order_assets() {
 		// Load Javascript for WooCommerce new order page
 		wp_enqueue_script( 'wc-taxjar-order', $this->wc_connect_base_url . 'woocommerce-services-new-order-taxjar-' . WC_Connect_Loader::get_wcs_version() . '.js', array( 'jquery' ), null, true );
 	}
+
+	/**
+	 * Add a 27 cent surcharge to your cart / checkout based on delivery country & state
+	 * Taxes, shipping costs and order subtotal are all included in the surcharge amount
+	 *
+	 * Fee won't be added if all products in the cart are virtual
+	 * 
+	 * Change $fee to set the surcharge to a value to suit
+	 *
+	 * 
+	 * Change $country to set the country
+	 * http://en.wikipedia.org/wiki/ISO_3166-1#Current_codes for available alpha-2 country codes
+	 *
+	 * Change $state to set the state
+	 * Example: The state code for Colorado would be 'CO'
+	 * https://en.wikipedia.org/wiki/ISO_3166-2:US
+	 *
+	 * Uses the WooCommerce fees API
+	 *
+	 * @param WC_Cart $cart WooCommerce Cart object.
+	 */
+	public function add_custom_state_surcharge_fee( $cart ) {
```

Comment:
We have to check if Colorado state Sales or Use Tax is charged on the items (see if we already have those taxes applied) as usually when items are exempt from state Sales or Use Tax they are exempt from Retail Delivery Fee.

Reply:
This is a nice catch. Thanks! I did try to dig into this and based on [this article](https://www.taxjar.com/blog/2022-08-what-is-the-colorado-retail-delivery-fee), and it said this : "If the entire retail sale is exempt from sales tax, the sale is also exempt from the retail delivery fee. If one item in the retail sale is exempt, but another item is subject to sales tax, the retailer should collect the retail delivery fee.". From WooCommerce point of view, it seems we don't need to apply this fee if either the sales tax is zero, or all products is non-taxable.

Reply:
Yes, if there is no tax on whole cart levelm no need to apply the fee.